VerifyProfileEmail/handler.ts

```diff
@@ -0,0 +1,232 @@
+import * as crypto from "crypto";
+
+import * as express from "express";
+
+import { isLeft } from "fp-ts/lib/Either";
+import { isNone } from "fp-ts/lib/Option";
+
+import * as t from "io-ts";
+
+import { Context } from "@azure/functions";
+import { TableService } from "azure-storage";
+
+import {
+  IResponseErrorValidation,
+  IResponsePermanentRedirect,
+  ResponsePermanentRedirect
+} from "italia-ts-commons/lib/responses";
+import { PatternString } from "italia-ts-commons/lib/strings";
+
+import { RequiredQueryParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_query_param";
+
+import { VerificationTokenEntity } from "io-functions-commons/dist/src/entities/verification_token";
+import { ProfileModel } from "io-functions-commons/dist/src/models/profile";
+import { retrieveTableEntity } from "io-functions-commons/dist/src/utils/azure_storage";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import { ValidUrl } from "italia-ts-commons/lib/url";
+
+// A token must be in the following format:
+// ULID + ":" + crypto.randomBytes(12)
+const TokenQueryParam = PatternString("^[A-Za-z0-9]{26}:[A-Fa-f0-9]{24}$");
+type TokenQueryParam = t.TypeOf<typeof TokenQueryParam>;
+
+type IVerifyProfileEmailHandler = (
+  context: Context,
+  token: string
+) => Promise<IResponsePermanentRedirect | IResponseErrorValidation>;
+
+/**
+ * Returns a ValidUrl in case of success
+ */
+function verificationSuccessUrl(verificationCallbackUrl: string): ValidUrl {
+  return { href: `${verificationCallbackUrl}?result=success` };
+}
+
+/**
+ * Returns a ValidUrl with a error in case of failure
+ */
+function verificationFailureUrl(
+  verificationCallbackUrl: string,
+  error: string
```

Comment:
is this a free text (in that case it should be encoded) or it's a well known string (in that case we need a more specific type)?